### PR TITLE
Fix security of nginx headers .

### PR DESCRIPTION
**Pull request by frju365**
# Problem

Security problems due to unsafe header : 
  - check : https://wiki.mozilla.org/Security/Guidelines/Web_Security
  - Test of the header : https://observatory.mozilla.org/ ; https://securityheaders.io/ ; https://hstspreload.org

# Solution

- Add preload to the HSTS header
- Add XSS protection (need CSP too)
- Add Referrer-Policy (need to be tesing for other applications) only in ADMIN conf.

# Improvement in future PR :

- Add OCSP Stapling
- Add HPKP-pining to SSL cert.
- Add referrer to server.tpl
- Improve CSP
- Add Report-URI header with proper PHP file in .well-known/report-uri/index.php 

# Status
Need to be testing.

### CODE REVIEW

**Comment by frju365** [comment]
Yes, it is needed to test it with several case and apps, to see if nothing is broken. For the referrer header, I will change it to 'same-origin', because 'origin-when-cross-origin' is not really safe (observatory of mozilla)

**Comment by Psycojoker** [comment]
Yeah, I think we need one full testing only for this patch to check that nothing has break :/

**Comment by frju365** [comment]
In fact the conf of the app is first. So, it won't break it. I need one testing at min for admin interface, and a test with at least 5 random apps(to be sure it won't break the app without conf overwriting).

**Comment by Psycojoker** [comment]
We can wait for a whole CI run on this PR no? Once it's merged, I remember we did that for something else.

**Comment by frju365** [comment]
For me, it's good, if someone test it ;) I have tested that with my app, and it seems to work.

**Comment by frju365** [comment]
I think that is good for old browser... But I will check that.

**Comment by frju365** [comment]
I wanted to improve it for the admin conf and perhaps add a true url for the report-uri CSP. Don't merge for instance.

File: data/templates/nginx/server.tpl.conf [hunk 1]
```diff
@@ -42,7 +42,14 @@ server {
     # > openssl dhparam -out /etc/ssl/private/dh2048.pem -outform PEM -2 2048
     #ssl_dhparam /etc/ssl/private/dh2048.pem;
 
-    add_header Strict-Transport-Security "max-age=31536000;";
+    add_header Strict-Transport-Security "max-age=63072000; includeSubDomains; preload"; 
+    add_header 'Referrer-Policy' 'same-origin';
```
Comment (zamentur) [inline data/templates/nginx/server.tpl.conf]: Need to be tested on django apps before to apply it.

Comment (zamentur) [reply data/templates/nginx/server.tpl.conf]: The feature to test on apps is the login feature of apps

Comment (frju365) [reply data/templates/nginx/server.tpl.conf]: Yes. for better testing I will split the PR in 2.

**Comment by frju365** [comment]
PR is now complete.

**Comment by frju365** [comment]
dans "Improvement in future PR", j'en ai profité pour noter les futures améliorations que je souhaiterais faire dans les header (et que j'ai déjà fait chez moi par la même occasion)

**Comment by Psycojoker** [comment]
Let's put that in a bit long testing then :)

**Pull request merged**